### ISSUE TITLE
On signup screen, "Already have a Signal Account?" is inaccessible on iPhone 6/6s

### ISSUE THREAD

**Issue by Empact** [signalapp/Signal-iOS#2321]
- [x] I have searched open and closed issues for duplicates

----------------------------------------

### Bug description
When I open the app, "Already have a Signal Account?" is covered by the numeric keypad for the phone number field. I'm unable to click out of the phone number field to dismiss the keypad. Clicking on the country code brings up an control that fills the window. I'm unable to click the "Already have a Signal Account?" link while switching between the controls.

### Steps to reproduce
- Install on iPhone 6
- Open the app
- Try to access "Already have a Signal Account?"

**Actual result:** the link is not accessible
**Expected result:** the link is accessible

### Screenshots
![img_8243](https://user-images.githubusercontent.com/5470/28136488-c5d8e942-66fe-11e7-8fdf-9125204a9d68.PNG)

### Device info
**Device**: iPhone 6
**iOS version**: 10.3.2
**Signal version:** 2.13.4

**Comment by resurrexit** [comment]
This is also inaccessible on:

**Device:** iPhone 6s
**iOS version:** 10.3.2
**Signal version:** 2.13.4

**Comment by pronoiac** [comment]
It's also unavailable on: 
**Device:** iPad Pro
**iOS version:** 10.3.3
**Singal version:** 2.15.3

This is important because I just blundered into a situation where I had to delete and reinstall Signal, losing my message archives, to regain functionality on my phone. That's not a minor UX issue.

**Comment by lippling** [comment]
Same here with an iPhone 8.

**Comment by benjuang** [comment]
Ack.  Just migrated to an iPhone 8 / iOS 11 and running into the same issue.

**Comment by ea396** [comment]
> This is important because I just blundered into a situation where I had to delete and reinstall Signal, losing my message archives, to regain functionality on my phone. That's not a minor UX issue.

I would also like to raise attention to this issue as I am currently locked out of my signal account and I am unable to do anything about it.
I feel forced to abandon all of my communication as well as access to group chats.

**Comment by pronoiac** [comment]
It's still an issue on Signal 2.16.2.

**Comment by maaku** [comment]
Same issue on iPhone 7 with iOS 11.

This is absolutely unacceptable. How has this issue been open for 3 months with zero developer activity? I am de facto locked out from my account and message history because of a UI issue. I’m looking for alternative apps.

**Comment by Empact** [comment]
@maaku Hey the project is open source so you could take a hand at fixing it yourself. Consider that the project is in many ways built on a desire to do public good and they have many things to attend to. Hats off to @moxie0!

**Comment by maaku** [comment]
I was under the mistaken impression that Open Whisper Systems was a company, not a non-profit, and presumed they sold b2b solutions or something. As I see now that is not the case, I retract the tone of my criticism but not the message. We carry around our lives on our communication devices, and those of us concerned about privacy keep that data silo'd inside the app. Pictures, notes, plans, arrangements, things in-flight .. all of it as good as gone. I'd be very sympathetic if this is a new issue, iOS 11 related or whatnot, but 2 1/2 months?

The change to fix this should be trivial, for someone who knows anything at all about iOS app development (which is not me) and who also is a registered Apple developer (not me). So lesson learned, I guess, and time to find a solution that won't hold my data hostage.

**Comment by hmhrex** [comment]
@maaku Patience is a virtue. I would love to fix this issue, but don't have the resources to do so at the moment. I'll be happy when it finally does get fixed. 😊

**Comment by IndieTea** [comment]
@maaku --- The issue of not being able to retrieve old data once you've deleted Signal is inherent to the idea and philosophy of Signal. The data (your old message history) is not being stored anywhere (thus it IS irretrievable, and cannot be subpoenaed or otherwise obtained by third parties). Even if Open Whisper Systems makes an accessible "Already have a Signal Account?" button, your issue with data storage will not be resolved.

**Comment by ssmereka** [comment]
Same issue on iPhone x

**Comment by Empact** [comment]
I think this is solved as of 2.19.2, and possibly earlier. Can anyone check and confirm? @ssmereka @maaku @pronoiac

**Comment by maaku** [comment]
@IndieTea I did not delete Signal. I did a phone restore from a backup file, which copied the old Signal data files onto the phone. I was able to verify that there was data storage used by the Signal app consistent in size with what I would have expected those messages to consume. Nevertheless when opening the Signal app after a restore for the first time, it presents the "Signup or Login" UX, presumably because the authentication token was not preserved, but the Login link was not accessible due to being blocked by the iPhone 7 keypad.

In any case I'm removing myself from this thread because I was forced to use the Signup option which wiped all prior messages on the device in the process (as well as corrupting session state, but that's another issue).

**Comment by pronoiac** [comment]
@Empact: using Signal 2.19.2 on my iPad Pro, no "Already have a Signal Account?" link is visible.

**Comment by Empact** [comment]
Yeah no such link for me either but I was able to go through the sign in
process and link my account with an existing Chrome Signal account.

The issue here was that there was no way to claim / activate an existing
account because the link was inaccessible, now there seems to be a way, if
not the old way.

**Comment by tonybrasunas** [comment]
Just experienced this migrating from iPhone6 -> iPhone8. Ugh. I think I'm wiped out.

@Empact Can you explain your solution a little more?

**Comment by DSouthworth** [comment]
Same issue on an iPhone SE

**Comment by Empact** [comment]
@moxie-signal Is it possible to get more attention on this?
